src/zkevm_specs/evm_circuit/execution/return_revert.py

```diff
@@ -24,36 +25,45 @@ def return_revert(instruction: Instruction):
 
     rwc_delta = 3
 
+    callee_gas_left = instruction.curr.gas_left
     if instruction.curr.is_create and is_success:
         # A. Returns the specified memory chunk as deployment code.
 
-        # TODO: Untested case.  Test it once create Tx is implemented, and once
-        # CREATE/CREATE2 are implemented.
         callee_address = instruction.call_context_lookup(CallContextFieldTag.CalleeAddress)
-        reversion_info = instruction.reversion_info()
         code_hash, code_hash_prev = instruction.account_write(
             callee_address, AccountFieldTag.CodeHash
         )
         instruction.constrain_equal(code_hash_prev, RLC(EMPTY_HASH))
         instruction.constrain_equal(code_hash, instruction.curr.aux_data)
 
+        # verify bytecode size less than 24,576 bytes
+        instruction.range_lookup(return_length, MAX_CODE_SIZE)
+
+        # gas cost of CREATE = GAS_COST_CREATE + memory expansion + GAS_COST_CODE_DEPOSIT * len(byte_code)
+        # first two part were handled in create.py
+        callee_gas_left = callee_gas_left - return_length * GAS_COST_CODE_DEPOSIT
+        instruction.constrain_equal(
+            instruction.call_context_lookup(CallContextFieldTag.GasLeft, RW.Write), callee_gas_left
+        )
```

Comment:
I think we don't need to update this again, since we already are going to do transition to caller and discard this call context.
```suggestion
```

Reply:
fixed in [d675ec9](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/355/commits/d675ec97ed971ec044a5bc5a40db3ab57551c629)